ci/test/05_before_script.sh

```diff
@@ -25,10 +25,15 @@ if [ -n "$XCODE_VERSION" ] && [ ! -f "$OSX_SDK_PATH" ]; then
 fi
 
 if [[ ${USE_MEMORY_SANITIZER} == "true" ]]; then
-  # Use BDB compiled using install_db4.sh script to work around linking issue when using BDB
+  # Use BDB compiled using install_db5.sh script to work around linking issue when using BDB
   # from depends. See https://github.com/bitcoin/bitcoin/pull/18288#discussion_r433189350 for
   # details.
-  DOCKER_EXEC "contrib/install_db4.sh \$(pwd) --enable-umrw CC=clang CXX=clang++ CFLAGS='${MSAN_FLAGS}' CXXFLAGS='${MSAN_AND_LIBCXX_FLAGS}'"
+  DOCKER_EXEC "contrib/install_db5.sh \$(pwd) --enable-umrw CC=clang CXX=clang++ CFLAGS='${MSAN_FLAGS}' CXXFLAGS='${MSAN_AND_LIBCXX_FLAGS}'"
+fi
+
+if [[ $HOST = *-apple-darwin16 ]]; then
+  # Use BDB compiled using install_db5.sh script as Brew does not supply BDB 5.3
+  DOCKER_EXEC "contrib/install_db5.sh \$(pwd)"
```

Comment:
locally, adding `--enable-posixmutexes` was all I needed to make this work on my M1:

```console
% clang --version
Apple clang version 12.0.5 (clang-1205.0.22.11)
Target: arm64-apple-darwin20.3.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

```suggestion
  DOCKER_EXEC "contrib/install_db5.sh \$(pwd) --enable-posixmutexes"
```

Reply:
That did seem a lot better, although now it's not finding the library, for reasons I can't readily see :-/

Reply:
I think that this is similar to an m4 thing I PR'd against #2745 - but in a different spot. I still have PR-ing you a fix on my todo. I started but had some issues with cirrus. Will come back to that.